pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,660 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicresources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	resourcev1alpha1 "k8s.io/api/resource/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	corev1ac "k8s.io/client-go/applyconfigurations/core/v1"
+	metav1ac "k8s.io/client-go/applyconfigurations/meta/v1"
+	resourcev1alpha1ac "k8s.io/client-go/applyconfigurations/resource/v1alpha1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	cgotesting "k8s.io/client-go/testing"
+	"k8s.io/klog/v2/ktesting"
+	_ "k8s.io/klog/v2/ktesting/init"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+)
+
+var (
+	podName       = "my-pod"
+	podUID        = types.UID("1234")
+	resourceName  = "my-resource"
+	resourceName2 = resourceName + "-2"
+	claimName     = podName + "-" + resourceName
+	claimName2    = podName + "-" + resourceName + "-2"
+	className     = "my-resource-class"
+	namespace     = "default"
+	templateName  = "my-template"
+
+	resourceClass = toResourceClass(resourcev1alpha1ac.ResourceClass(className))
+
+	podWithClaimName      = toPod(corev1ac.Pod(podName, namespace).WithUID(podUID).WithSpec(corev1ac.PodSpec().WithResourceClaims(corev1ac.PodResourceClaim().WithName(resourceName).WithSource(corev1ac.ClaimSource().WithResourceClaimName(claimName)))))
```

Comment:
Finally had a chance to review this file.

Remove those helpers.
You can use `pkg/scheduler/testing`

Reply:
That doesn't have helper functions for any of new resourcev1allphav1 types. My hope was that by using the generated applyconfiguration functions I would not have to write the corresponding helper functions by hand. Are you saying that I should not use applyconfiguration?

Reply:
It looks odd, combined with marshalling/unmarshalling.

I prefer we use the pattern that we already established of having a builder.

Reply:
So the `toPod` and `toResourceClass` calls are the odd part, right?

I would like to investigate how to get rid of those: https://github.com/kubernetes/kubernetes/issues/113773

In the meantime, can we keep this file as it is? If really don't want it, then I can still change it before test freeze.

Reply:
I replaced the usage of applyconfiguration with some additional helpers in `pkg/scheduler/testing`. The use of generics is also gone.

No changes to test cases and code coverage yet. I'll work on that next.